vignettes/odin_models.Rmd

```diff
@@ -0,0 +1,196 @@
+Where $S$ is the number of susceptibles, $I$ is the number of infected and $R$ is the number recovered; the total population size $N = S + I + R$ is constant. $\beta$ is the infection rate, $\gamma$ is the recovery rate.
+
+We can write the equivalent deterministic odin model as follows:
+
+```{r}
+sir_code <- "initial(S) <- N_init - I_init
+             initial(I) <- I_init
+             initial(R) <- 0
+             initial(N) <- N_init
+             
+             deriv(S) <- -beta * S * I / N
+             deriv(I) <- beta * S * I / N - gamma * I
+             deriv(R) <- gamma * I
+             deriv(N) <- 0
+             
+             beta <- user(0.4)
+             gamma <- user(0.2)
+             N_init <- user(1e5)
+             I_init <- user(1)"
+```
+
+
+By compiling this model code with `odin.dust` rather than `odin` we can use the calibration functions in mcstate
+
+```{r}
+gen_sir <- odin.dust::odin_dust(sir_code)
+```
+
+We will attempt to infer parameters $\beta$ and $\gamma$ using weekly measures of population prevalence, for example taken from household prevalence surveys. These have been generated using $\beta = 0.3$ and $\gamma = 0.1$, assuming around 100 people are sampled each week.
+
+```{r}
+data <- read.csv(system.file("sir_prevalence.csv", package = "mcstate"))
+
+par(bty = "n", mar = c(3, 3, 1, 1), mgp = c(1.5, 0.5, 0))
+test_cols <- c(Tested = "darkorange2", Positive = "green4")
+plot(data$day, data$n_tested, type = "l",
+     xlab = "Day", ylab = "Number of individuals",
+     ylim = c(0, max(data$n_tested)), col = test_cols[1])
+lines(data$day, data$n_positive, type = "l", col = test_cols[2])
+legend("left", names(test_cols), fill = test_cols, bty = "n")
+```
+We first need to process the data into the `mcstate` format. As the `rate` parameter is not meaningful in the context of a continuous time model, we supply a `NULL` value.
+
+```{r}
+sir_data <- mcstate::particle_filter_data(data = data, time = "day",
+                                          initial_time = 0, rate = NULL)
+
+rmarkdown::paged_table(sir_data)
+```
+
+We define a simple index function for our model, to provide meaningful parameter names: 
+
+```{r}
+index <- function(info) {
+  idx <- unlist(info$index)
+  list(run = idx, state = idx)
+}
+```
+
+Next, we define the comparison function to calculate the likelihood of the simulation given the data, which in our case will be a Binomial distribution.
+
+```{r}
+compare <- function(state, observed, pars = NULL) {
+  prevalence <- state["I", , drop = TRUE] / state["N", , drop = TRUE]
+  dbinom(x = observed$n_positive, size = observed$n_tested, prob = prevalence,
+         log = TRUE)
+}
+```
+
+Although we are fitting a deterministic model, to use `mcstate` functionality we still need to define a particle filter object. Running this filter will give us the same output as running the deterministic model:
+
+```{r}
+
+filter <- mcstate::particle_deterministic$new(data = sir_data, model = gen_sir,
```

Comment:
this does need a different name; this one was only ever a placeholder that has got stuck. Suggestions on the back of a postcard please